scopesim/effects/detector_list.py

```diff
@@ -151,48 +163,104 @@ def apply_to(self, obj, **kwargs):
         if not isinstance(obj, FovVolumeList):
             return obj
 
-        hdr = self.image_plane_header
-        xy_mm = calc_footprint(hdr, "D")
-        pixel_size = hdr["CDELT1D"]  # mm
-        pixel_scale = kwargs.get("pixel_scale", self.meta["pixel_scale"])  # ["]
-        pixel_scale = from_currsys(pixel_scale, self.cmds)
+        shrink_axis, shrink_values = self._get_fov_limits(**kwargs)
+        obj.shrink(axis=shrink_axis, values=shrink_values)
+
+        return obj
 
-        # x["] = x[mm] * ["] / [mm]
-        xy_sky = xy_mm * pixel_scale / pixel_size
+    @property
+    def pixel_size(self) -> u.Quantity[u.mm]:
+        """Return size of one pixel in mm."""
+        pixel_size = np.min(
+            quantity_from_table("pixel_size", self.active_table, u.mm))
+        return pixel_size
 
-        obj.shrink(axis=["x", "y"],
-                   values=(tuple(zip(xy_sky.min(axis=0),
-                                     xy_sky.max(axis=0)))))
+    @property
+    def pixel_scale_mm(self) -> u.Equivalency:
+        """Return pixel scale (mm / pix) as equivalency."""
+        return u.pixel_scale(self.pixel_size / u.pixel)
+
+    def _get_pixel_scale_arcsec(self, **kwargs) -> u.Equivalency:
+        """To allow overriding defaults, but use defaults in property."""
+        pixel_scale = u.pixel_scale(
+            from_currsys(
+                kwargs.get("pixel_scale", self.meta["pixel_scale"]),
+                self.cmds,
+            ) << u.arcsec / u.pixel
+        )
+        return pixel_scale
```

Comment:
Please, please reduce the number of `**kwargs` flowing around. There is no reason for any of those
```suggestion
    def _get_pixel_scale_arcsec(self, pixel_scale=None) -> u.Equivalency:
        """To allow overriding defaults, but use defaults in property."""
        pixel_scale = u.pixel_scale(
            from_currsys(
                pixel_scale if pixel_scale is not None else self.meta["pixel_scale"],
                self.cmds,
            ) << u.arcsec / u.pixel
        )
        return pixel_scale
```

Reply:
Yeah, you're right. Should the `self._get_fov_limits(**kwargs)` in `apply_to()` (where the `pixel_scale` would ultimately come in) be changed to something like `self._get_fov_limits(pixel_scale=kwargs.get("pixel_scale"))`? Otherwise we might get `TypeError: _get_fov_limits() got an unexpected keyword argument` if `kwargs` contains anything else, right?

Reply:
Also, I'm not even sure if the `pixel_scale` is ever actually passed along this way "in the wild". Isn't it usually always given through the cmds anyway?

Reply:
We can slowly remove things from `**kwargs`. So `apply_to()` can become something like
```
def apply_to(self, pixel_scale, **kwargs):
```
and if we need to `**kwargs` along later we can do
```
self.some_other_function(pixel_scale=pixel_scale, **kwargs)
```
that is, make everything as explicit as possible, and only use `**kwargs` when we need to pass something along but we don't know what a-priori.